android/src/main/kotlin/hk/gogovan/flutter_label_printer/searcher/UsbSearcher.kt

```diff
@@ -45,9 +46,31 @@ class UsbSearcher(private val context: Context) {
         }
     }
 
-    fun getUsbDevices(): Map<String, UsbDevice> {
+    suspend fun getUsbDevices(): List<UsbDevice> {
+        val result = mutableListOf<UsbDevice>()
+        var permissionPendingChecks = 0
+
         val manager = context.getSystemService(UsbManager::class.java)
-        return manager.deviceList
+        for (device in manager.deviceList.values) {
+            for (i in 0 until device.interfaceCount) {
+                val intf = device.getInterface(i)
+                if (intf.interfaceClass == 7) {
```

Comment:
@luiges90 what does the `7` mean? better to use constant or enum

Reply:
This is copied from the sample code of Hanin printer, but I think I found the definition in Android docs, let me update it...